### PR TITLE
Add jupyter/julia-notebook

### PR DESCRIPTION
**Pull request by yuvipanda**
## Describe your changes

There is a growing number of Julia users in the Jupyter ecosystem who do not use R, and hence would <3 to have a dedicated docker image that doesn't bring in all the R stuff that datascience-notebook brings in! The built image size is much smaller, and eventually paves the way to better ecosystem support for Julia.

## Checklist (especially for first-time contributors)

- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] I will try not to use force-push to make the review process easier for reviewers
- [ ] I have updated the documentation for significant changes

### CODE REVIEW

**Comment by yuvipanda** [comment]
This image is  2.2GB, vs the datascience-notebook being 5.83GB!

**Comment by mathbunnyru** [comment]
Docs failing is fine.
`Pre-commit` mypy error is valid though:
```
tests/julia-notebook/test_julia.py: error: Duplicate module named "test_julia" (also at "tests/datascience-notebook/test_julia.py"
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -26,6 +26,7 @@ body:
         - minimal-notebook
         - pyspark-notebook
         - r-notebook
+        - julia-notebook
```
Comment (mathbunnyru) [inline .github/ISSUE_TEMPLATE/bug_report.yml:29]: Let's keep this list sorted.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 2]
```diff
@@ -24,6 +24,7 @@ body:
         - minimal-notebook
         - pyspark-notebook
         - r-notebook
+        - julia-notebook
```
Comment (mathbunnyru) [inline .github/ISSUE_TEMPLATE/feature_request.yml:27]: This list is sorted as well.

File: julia-notebook/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Jupyter Notebook Data Science Stack
```
Comment (mathbunnyru) [inline julia-notebook/README.md:1]: Name has to be changed here.

**Review by mathbunnyru — CHANGES_REQUESTED** [review]
There are a few things I would like to ask you to do:

1. In places, where images were sorted by name, let's keep them sorted.
2. In other places, let's put `julia-notebook` after `r-notebook`.
3. Please, update the [Makefile](https://github.com/jupyter/docker-stacks/blob/main/Makefile#L11)
4. And also, [inheritance diagram](https://github.com/jupyter/docker-stacks/blob/main/docs/using/selecting.md#image-relationships)

I also think, that it makes sense to put Julia installation in a bash script and to add it to `base-notebook` and then we will just run it in some images. This will allow us to reduce code duplication. This can be done in a separate PR later though, to make it easier for you.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
Comment (yuvipanda) [reply .github/ISSUE_TEMPLATE/bug_report.yml:29]: Done!

File: julia-notebook/README.md (hunk 3 above)
Comment (yuvipanda) [reply julia-notebook/README.md]: Done

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 2 above)
Comment (yuvipanda) [reply .github/ISSUE_TEMPLATE/feature_request.yml]: Done!

**Comment by yuvipanda** [comment]
Thanks for the quick review, @mathbunnyru! I have addressed everything except the inheritance diagram! Do you have a tool suggestion on what I can use to edit that?

**Comment by mathbunnyru** [comment]
@yuvipanda take a look at the markdown, it's not just an SVG image, but also a link where you can change the diagram easily, generate new permanent link and download an image as a new SVG.
<http://interactive.blockdiag.com/?compression=deflate&src=eJyFjjFuwzAMRXefgvDUDDpAYaQnyNaOBQraphPCCmlIFIy0yN0rdQggI4XX9x__Z-91mEfGM_w0ACNNmLx9TSoW-ZvgCK9d5hqYxNBYJaNFgwVk65octalPYulTXk4f77CyXXLOYhDIE0Y6tODeYMwrFFw0HOboJk0y_rWV8v-yctfnBidq1KvORa5AMa4sfEVfSVtWvDjwcqusmhSnNZKoYfK6Pnj7XM0vYqYkA-1Oh53Z5RYXDHNlbVnx0Hu3Ne-_7MWcxg>

**Comment by yuvipanda** [comment]
That's very helpful, @mathbunnyru! I think I've addressed them now.

**Comment by mathbunnyru** [comment]
@yuvipanda I added a new repository to the DockerHub and merged your PR (that's all I had to do manually, even description will be pushed automatically to DockerHub 🙂).

I didn't congratulate you with the first PR to this repo, so let me congratulate you with the second one :heart:

**Comment by mathbunnyru** [comment]
https://hub.docker.com/r/jupyter/julia-notebook

Images should be updated in ~2 hours.

**Comment by yuvipanda** [comment]
AMAZING, THANK YOU SO MUCH @mathbunnyru!

**Comment by mathbunnyru** [comment]
> AMAZING, THANK YOU SO MUCH @mathbunnyru!

You're welcome:)

If you have some spare time, it would be great to have a Julia installation script put to `jupyter/minimal-notebook`, so we don't duplicate big chunks of code in Dockerfiles.

File: .github/workflows/docker.yml [hunk 4]
```diff
@@ -254,6 +275,7 @@ jobs:
             minimal-notebook,
             scipy-notebook,
             r-notebook,
+            julia-notebook,
```
Comment (mathbunnyru) [inline .github/workflows/docker.yml:278]: We forgot to add a new `julia` dependency in `needs:` in a few lines below.

File: .github/workflows/docker.yml [hunk 5]
```diff
@@ -290,6 +312,7 @@ jobs:
...
```
Comment (mathbunnyru) [inline .github/workflows/docker.yml:315]: We forgot to add a new `julia` dependency in `needs:` in a few lines below.

**Comment by mathbunnyru** [comment]
@yuvipanda I left these comments because I use this image as an example of how a new image can be added.
If you can, please cherry-pick this commit to your branch (so, this PR will be 100% correct).
If not, just ignore this message 🙂

**Comment by yuvipanda** [comment]
@mathbunnyru that's so sweet! I've cherry-picked that commit but it doesn't seem to be reflected here.

**Comment by mathbunnyru** [comment]
Thanks @yuvipanda
It seems to be GitHub limitations, so we can't do anything here, unfortunately.
Thanks for trying 🙂

**Pull request merged**